### ISSUE TITLE
&ft not set in startup

### ISSUE THREAD

**Issue by TornaxO7** [neovim/neovim#14437]
- `nvim --version`:

<details>

```vim
NVIM v0.5.0-dev+1176-g9481310c2
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/cc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/tornax/.config/yay/build/neovim-nightly-git/src/build/config -I/home/tornax/.config/yay/build/neovim-nightly-git/src/neovim-nightly-git/src -I/usr/include -I/home/tornax/.config/yay/build/neovim-nightly-git/src/build/src/nvim/auto -I/home/tornax/.config/yay/build/neovim-nightly-git/src/build/include
Compiled by tornax

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

</details>

- `vim -u DEFAULTS` (version: ) behaves differently? - No
- Operating system/version: Arch Linux, Kernel: 5.11.16-arch1-1
- Terminal name/version: alacritty 0.7.2 (5ac8060b)
- `$TERM`: alacritty

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
# Alternative for shell-related problems:
# env -i TERM=ansi-256color "$(which nvim)"
```

1. Put this into your `init.vim`:
```vim
echom 'Filetype: ' . &ft
```
2. Open any file

### Actual behaviour
I'm getting this output:
(Opening init.vim)
![image](https://user-images.githubusercontent.com/50843046/115994504-8bfb8380-a5d7-11eb-8f38-f7b281a8d0a8.png)

Opening test.c:
![image](https://user-images.githubusercontent.com/50843046/115994516-9a499f80-a5d7-11eb-9fe5-aa83bd8656fb.png)

### Expected behaviour
Output: `Filetype: <filetype>` for example `Filetype: c`

**Comment by teto** [comment]
your reproduction instructions is weird, if you launch with NORC, how do you execute the command ? if it's executed before launching any file or with empty buffer `:enew` than there is no filetype so it is expected. If you run the command once the filetype is set it works.

**Comment by TornaxO7** [comment]
Here's a demo:

https://user-images.githubusercontent.com/50843046/115995606-34134b80-a5dc-11eb-8c73-13e978b7455e.mp4

When I startup vim `&ft` is nothing. I thought that it should be set from the beginning?

**Comment by mhinz** [comment]
That's expected behaviour. When you source a file, it gets executed as if it was typed in the command-line.

When you start Nvim, you end up in an empty scratch buffer with no filetype. That's why you see no output.

**Comment by TornaxO7** [comment]
But I didn't open an unnamed buffer? I opened a vim file? So shoudn't neovim display `Filetype: vim` at the bottom?

**Comment by mhinz** [comment]
What's the exact command you used?

Mind that the vimrc is the first thing to be sourced (among many others) and only then the file given on the commandline is opened.

**Comment by TornaxO7** [comment]
Have you watched my video? Shouldn't the command `echom 'Filetype: ' . &ft` be executed if I open nvim?

**Comment by mhinz** [comment]
As I said above, by the time the `echom 'Filetype: ' . &ft` is run during initialization, there is no file opened yet and thus no filetype set.

**Comment by TornaxO7** [comment]
Hm... ok than I'll have to find another solution. Thank you!

**Comment by mhinz** [comment]
What's the actual problem? What do you want to do?

**Comment by TornaxO7** [comment]
I want to write a little [plugin](https://github.com/TornaxO7/auto-cosco.vim) which looks if the current filetype is in the whitelist. Currently I put this into the `plugin/cosco.vim` file:
```vim
if index(g:cosco_whitelist, &ft) == -1
    let g:cosco_enable = 0
    echom 'filetype: ' . &ft
endif
```
I was wondering why `index(g:cosco_whitelist, &ft)` returned -1 for each file I opened and I saw that `&ft` is empty every time. Is there a better solution than creating files for each filetype in  `ftplugin`?
